mLRS/Common/arq.h

```diff
@@ -0,0 +1,368 @@
+{
+    status = ARQ_RX_IDLE;
+    received_seq_no = 0;
+    received_seq_no_last = 0;
+    accept_received_payload = false;
+    frame_lost = false;
+    ack_seq_no = 0;
+}
+
+
+// methods called upon receive or expected receive (in doPreTransmit)
+// the calling sequence is:
+// 1. Received(seq_no) or FrameMissed() (in handle_receive() or handle_receive_none())
+// 2. AcceptPayload() (in process_received_frame())
+// 3. FrameLost()
+// 4. Disconnected()
+
+void tReceiveArq::Disconnected(void)
+{
+    status = ARQ_RX_IDLE;
+}
+
+
+void tReceiveArq::spin(void)
+{
+    if (status == ARQ_RX_IDLE) { while(1); } // must not happen, should have been called after Missed,Received
+
+    switch (status) {
+    case ARQ_RX_RECEIVED_WAS_IDLE:
+        accept_received_payload = true;
+        received_seq_no_last = received_seq_no;
+        ack_seq_no = received_seq_no;
+        break;
+
+    case ARQ_RX_RECEIVED:
+        accept_received_payload = (received_seq_no != received_seq_no_last); // new seq no received, so accept it
+        received_seq_no_last = received_seq_no;
+        ack_seq_no = received_seq_no;
+        break;
+
+    case ARQ_RX_FRAME_MISSED: // is not called if handle_receive_none(), RX_STATUS_NONE !
+        accept_received_payload = false;
+        break;
+
+    default:
+        accept_received_payload = false;
+    }
+
+    frame_lost = false;
+}
+
+
+void tReceiveArq::FrameMissed(void)
+{
+    status = ARQ_RX_FRAME_MISSED;
+    spin();
+}
+
+
+void tReceiveArq::Received(uint8_t seq_no)
+{
+    received_seq_no = seq_no;
+    status = (status == ARQ_RX_IDLE) ? ARQ_RX_RECEIVED_WAS_IDLE : ARQ_RX_RECEIVED;
+    spin();
+}
+
+
+bool tReceiveArq::AcceptPayload(void)
+{
+    return accept_received_payload;
+}
+
+
+bool tReceiveArq::FrameLost(void) // called to check if parsers need to be reset
+{
+    return frame_lost;
+}
+
+
+// methods called for transmit
+
+uint8_t tReceiveArq::AckSeqNo(void)
+{
+    return ack_seq_no; // will be converted by 1 bit to 0/1
+}
+
+
+// miscellaneous
+
+bool tReceiveArq::SimulateMiss(void)
+{
+#if USE_ARQ_TX_SIM_MISS
+    static uint8_t miss_cnt = 0;
+    DECc(miss_cnt, USE_ARQ_TX_SIM_MISS);
+    if (miss_cnt == 0) return true;
+#endif
+    return false;
+}
+
+
+#else
+// these should be nfc, i.e., result in exactly the same behavior as before
+
+class tTransmitArq
+{
+  public:
+    void Init(void) { seq_no = 0; }
+
+    void Disconnected(void) {}
+    void FrameMissed(void) {}
+    void Received(uint8_t ack_seq_no) {}
+
+    bool GetFreshPayload(void) { return true; }
+    uint8_t SeqNo(void) { seq_no++; return seq_no; }
+    void SetRetryCnt(uint8_t retry_cnt) {}
+    void SetRetryCntAuto(int32_t _frame_cnt) {}
+
+    uint8_t seq_no;
+};
+
+
+class tReceiveArq
+{
+  public:
+    void Init(void) {}
+
+    void Disconnected(void) {}
+    void FrameMissed(void) {}
+    void Received(uint8_t _seq_no) {}
+    bool AcceptPayload(void) { return true; }
+    bool FrameLost(void) { return false; }
```

Comment:
oh ... hmhmhm ... this would need we do this fully ... hmhmhm 
you are right that this isn't restoring the original behavior
this piece of code should actually go once we decide to merge arq
I guess the easiest to do is to remove that function and in the main code to have define swicth restoring the old line of code if USE_ARQ is not defined